### PR TITLE
Pr 8341 and payload len fix/v4

### PR DESCRIPTION
**Pull request by inashivb**
- https://github.com/OISF/suricata/pull/8404
- https://github.com/OISF/suricata/pull/8408

suricata-verify-pr: 1077

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 128237 | 103871 | 81.0% |
| .app_layer.error.tls.parser | 931 | 1145 | 122.99% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.http | 3118951 | 3340797 | 107.11% |
| .app_layer.flow.tls | 510462 | 623355 | 122.12% |
| .app_layer.flow.ssh | 8937 | 11032 | 123.44% |
| .app_layer.flow.telnet | 1624 | 2334 | 143.72% |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 1160775 | 0.20 | 

Pipeline 11792

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.tls.parser | 931 | 1145 | 122.99% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.flow.http | 3118951 | 3340803 | 107.11% |
| .app_layer.flow.tls | 510462 | 623355 | 122.12% |
| .app_layer.flow.ssh | 8937 | 11032 | 123.44% |
| .app_layer.flow.telnet | 1624 | 2334 | 143.72% |
| .app_layer.error.ftp-data.gap | 0 | 1 | - |
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 3014 | 0.00 | 

Pipeline 11817

File: src/detect-engine-event.c [hunk 1]
```diff
@@ -161,6 +169,16 @@ static DetectEngineEventData *DetectEngineEventParse (const char *rawstr)
     if (de->event == STREAM_REASSEMBLY_OVERLAP_DIFFERENT_DATA) {
         StreamTcpReassembleConfigEnableOverlapCheck();
     }
+
+    if (OutdatedEvent(rawstr)) {
+        if (SigMatchStrictEnabled(DETECT_DECODE_EVENT)) {
+            SCLogError("decode-event keyword no longer supports event \"%s\"", rawstr);
```
Comment (victorjulien) [inline src/detect-engine-event.c:175]: need an SV test for this case

File: src/detect-engine-event.c [hunk 2]
```diff
@@ -161,6 +169,16 @@ static DetectEngineEventData *DetectEngineEventParse (const char *rawstr)
...
+            goto error;
+        } else {
+            SCLogWarning("decode-event keyword no longer supports event \"%s\"", rawstr);
```
Comment (victorjulien) [inline src/detect-engine-event.c:178]: need an SV test for this case

File: rules/decoder-events.rules [hunk 3]
```diff
@@ -67,7 +67,7 @@ alert pkthdr any any -> any any (msg:"SURICATA TCP option invalid length"; decod
 alert pkthdr any any -> any any (msg:"SURICATA TCP duplicated option"; decode-event:tcp.opt_duplicate; classtype:protocol-command-decode; sid:2200037; rev:2;)
 alert pkthdr any any -> any any (msg:"SURICATA UDP packet too small"; decode-event:udp.pkt_too_small; classtype:protocol-command-decode; sid:2200038; rev:2;)
 alert pkthdr any any -> any any (msg:"SURICATA UDP header length too small"; decode-event:udp.hlen_too_small; classtype:protocol-command-decode; sid:2200039; rev:2;)
-alert pkthdr any any -> any any (msg:"SURICATA UDP invalid header length"; decode-event:udp.hlen_invalid; classtype:protocol-command-decode; sid:2200040; rev:2;)
+alert pkthdr any any -> any any (msg:"SURICATA UDP invalid length field in the header"; decode-event:udp.len_invalid; classtype:protocol-command-decode; sid:2200220; rev:2;)
```
Comment (victorjulien) [inline rules/decoder-events.rules:70]: need an SV test that shows this matches

**Comment by victorjulien** [comment]
Please open a new PR instead of force pushing.

**Pull request closed without merging**